extensions/cms/src/cmsResource/tree/cmsResourceTreeNode.ts

```diff
@@ -38,12 +39,20 @@ export class CmsResourceTreeNode extends CmsResourceTreeNodeBase {
 		try {
 			let nodes: CmsResourceTreeNodeBase[] = [];
 			if (!this.ownerUri) {
-				this._ownerUri = await this.appContext.cmsUtils.getUriForConnection(this.connection);
+				// Set back password to get ownerUri
+				if (this.connection.options.authenticationType === 'SqlLogin' && this.connection.options.savePassword === true) {
+					this.connection.options.password = await this.appContext.cmsUtils.getPassword(this.connection.options.user);
+				}
 			}
 			return this.appContext.cmsUtils.createCmsServer(this.connection, this.name, this.description).then((result) => {
 				if (result) {
-					if (result.registeredServersList) {
-						result.registeredServersList.forEach((registeredServer) => {
+					// cache new connection is different from old one
+					if (this.appContext.cmsUtils.didConnectionChange(this._connection, result.connection)) {
+						this._connection = result.connection;
```

Comment:
@Charles-Gagnon Here, we set the connection for this node from the result. And we get the ownerUri from the connection, so that's how ownerUri gets set.

Reply:
But you're passing in this.ownerUri to other places (such as the RegisteredServerTreeNode constructor below) - and with your changes that's not going to have the correct value...

---
In reply to: [295078933](https://github.com/microsoft/azuredatastudio/pull/5989#discussion_r295078933) [](ancestors = 295078933)

Reply:
The child nodes are basically nodes with payload so they don't need an ownerUri. I set it anyway though.